core/src/main/java/org/apache/calcite/rel/rules/materialize/MaterializedViewRule.java

```diff
@@ -1252,6 +1285,193 @@ protected RexNode replaceWithOriginalReferences(final RexBuilder rexBuilder,
+        if (!(r.lowerEndpoint() instanceof TimestampString)
+            || !(r.upperEndpoint() instanceof TimestampString)) {
+          // ignore predicates where the type isn't a timestamp
+          return rexBuilder.makeLiteral(true);
+        }
+        RexNode lb = rexBuilder.makeTimestampLiteral((TimestampString) r.lowerEndpoint(), 0);
+        // if the lower bound is exclusive we need to shift the modified value to the next higher
+        // floored value
+        long modifiedLbVal = getModifiedVal(r.lowerBoundType() == BoundType.OPEN, true, lb);
+        TimestampString lbTimestampString = TimestampString.fromMillisSinceEpoch(modifiedLbVal);
+        RexNode ub = rexBuilder.makeTimestampLiteral((TimestampString) r.upperEndpoint(), 0);
+        // if the upper bound is inclusive we need to shift the modified value to the previous lower
+        // floored value
+        long modifiedUbVal = getModifiedVal(r.upperBoundType() == BoundType.CLOSED, false, ub);
+        TimestampString ubTimestampString = TimestampString.fromMillisSinceEpoch(modifiedUbVal);
+
+        lowerBound = modifiedLbVal;
+        upperBound = modifiedUbVal;
+
+        if (generateViewFilter) {
+          tableInputRefOperand = rexInputRef;
+        }
+
+        RelDataType type = call.getOperands().get(1).getType();
+        Sarg modSarg = Sarg.of(RexUnknownAs.UNKNOWN,
+            ImmutableRangeSet.of(Range.closedOpen(lbTimestampString, ubTimestampString)));
+        RexNode searchArgumentLiteral = rexBuilder.makeSearchArgumentLiteral(modSarg, type);
+        RexNode res = rexBuilder.makeCall(SqlStdOperatorTable.SEARCH, tableInputRefOperand,
+            searchArgumentLiteral);
+        return res;
+      // Since the view contains a FLOOR() we use GREATER_THAN_OR_EQUAL for the lower bound
+      // range comparison
+      case GREATER_THAN_OR_EQUAL:
+      case GREATER_THAN:
+        // CEIL is used to determine the lower bound range that for the view predicate
+        // For eg if the query has (date > TIMESTAMP'2018-01-01 01:00:00') and we have a view
+        // that rolls up to the minute we generate a predicate (date >= TIMESTAMP'2018-01-01
+        // 02:00:00')
+        return transformCall(call, true);
+      // since the view contains a FLOOR() we use LESS_THAN for the upper bound
+      // range comparison
+      case LESS_THAN:
+      case LESS_THAN_OR_EQUAL:
+        // FLOOR is used to determine the upper bound range that for the view predicate
+        // For eg if the query has (date <= TIMESTAMP'2018-01-01 05:00:00') and we have a view
+        // that rolls up to the minute we generate a predicate (date < TIMESTAMP'2018-01-01
+        // 04:00:00')
+        return transformCall(call, false);
+      case EQUALS:
+        return rexBuilder.makeLiteral(true);
+      default:
+        return super.visitCall(call);
+      }
+    }
+
+    public boolean isRangeMatched() {
+      return lowerBound != 0L || upperBound != 0L && lowerBound != upperBound;
```

Comment:
Don't you mean the following?

```suggestion
      return (lowerBound != 0L || upperBound != 0L) && lowerBound != upperBound;
```

Reply:
fixed